strutil/set_test.go

```diff
@@ -0,0 +1,74 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package strutil_test
+
+import (
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/strutil"
+)
+
+type orderedSetSuite struct {
+	set strutil.OrderedSet
+}
+
+var _ = Suite(&orderedSetSuite{})
+
+func (s *orderedSetSuite) SetUpTest(c *C) {
+	s.set = strutil.OrderedSet{}
+}
+
+func (s *orderedSetSuite) TestZeroValueItems(c *C) {
+	c.Assert(s.set.Items(), HasLen, 0)
+}
+
+func (s *orderedSetSuite) TestZeroValueContains(c *C) {
+	c.Check(s.set.Contains("foo"), Equals, false)
+}
+
+func (s *orderedSetSuite) TestZeroValueIndexOf(c *C) {
+	c.Check(s.set.Contains("foo"), Equals, false)
+}
+
+func (s *orderedSetSuite) TestZeroValuePut(c *C) {
+	s.set.Put("foo")
+	c.Check(s.set.Contains("foo"), Equals, true)
+	c.Check(s.set.Items(), DeepEquals, []string{"foo"})
+	idx, ok := s.set.IndexOf("foo")
+	c.Check(ok, Equals, true)
+	c.Check(idx, Equals, 0)
```

Comment:
I think it would be good to add at least one more string to get a non-zero index in the test.

Reply:
Maybe something like:
```
func (s *orderedSetSuite) TestZeroValuePut(c *C) {
	strs := []string{"foo", "bar", "baz"}
	for i, n := range strs {
		s.set.Put(n)
		c.Check(s.set.Contains(n), Equals, true)
		c.Check(s.set.Items(), DeepEquals, strs[:i+1])
		idx, ok := s.set.IndexOf(n)
		c.Check(ok, Equals, true)
		c.Check(idx, Equals, i)
                c.Check(s.set.Size(), Equals, i+1)
	}
}
``` 
?

Reply:
I wrote a test modeled after this one. Thanks :)